devito/ir/support/basic.py

```diff
@@ -1,260 +1,14 @@
-from collections import OrderedDict
-
 from cached_property import cached_property
-from sympy import Basic, S
+from sympy import S
 
 from devito.ir.support.space import Any, Backward
+from devito.ir.support.vector import LabeledVector, Vector
 from devito.symbolics import retrieve_terminals, q_monoaffine
 from devito.tools import (EnrichedTuple, Tag, as_tuple, is_integer,
                           filter_sorted, flatten, memoized_meth)
 from devito.types import Dimension
 
-__all__ = ['Vector', 'LabeledVector', 'IterationInstance', 'Access',
-           'TimedAccess', 'Scope']
-
-
-class Vector(tuple):
```

Comment:
note for reviewers: this entire deleted block was actually simply moved to vector.py